adapters/tlsio_ssl_esp8266.c

```diff
@@ -0,0 +1,712 @@
+
+    do {
+        ret = netconn_gethostbyname(tls_io_instance->hostname, &target_ip);
+    } while(ret);
+    // This is useful for debugging purpose.
+    // LogInfo("get target IP is %d.%d.%d.%d\n", (unsigned char)((target_ip.addr & 0x000000ff) >> 0),
+    //                                             (unsigned char)((target_ip.addr & 0x0000ff00) >> 8),
+    //                                             (unsigned char)((target_ip.addr & 0x00ff0000) >> 16),
+    //                                             (unsigned char)((target_ip.addr & 0xff000000) >> 24));
+
+    LogInfo("create socket ......");
+    sock = socket(AF_INET, SOCK_STREAM, 0);
+    if (sock < 0) {
+        LogError("create socket failed\n");
+        return;
+    }
+    LogInfo("create socket OK\n");
+
+    LogInfo("bind socket ......");
+    memset(&sock_addr, 0, sizeof(sock_addr));
+    sock_addr.sin_family = AF_INET;
+    sock_addr.sin_addr.s_addr = 0;
+    sock_addr.sin_port = htons(OPENSSL_LOCAL_TCP_PORT);
+    ret = bind(sock, (struct sockaddr*)&sock_addr, sizeof(sock_addr));
+    if (ret) {
+        LogError("bind socket failed\n");
+        return;
+    }
+    LogInfo("bind socket OK\n");
+
+    LogInfo("socket connect to remote ......");
+    memset(&sock_addr, 0, sizeof(sock_addr));
+    sock_addr.sin_family = AF_INET;
+    sock_addr.sin_addr.s_addr = target_ip.addr;
+    sock_addr.sin_port = htons(tls_io_instance->port);
+
+    lwip_set_non_block(sock);
+
+    ret = connect(sock, (struct sockaddr*)&sock_addr, sizeof(sock_addr));
+    if (ret == -1) {
+        ret = lwip_net_errno(sock);
+        if (ret != EINPROGRESS){
+            LogError("socket connect failed %s\n", tls_io_instance->hostname);
+            return;
+        } else{
+            LogInfo("socket connect in progress\n");
+        }
+    }
+    LogInfo("socket connect OK\n");
+
+    char recv_buf[128];
+    int recv_bytes = (int)sizeof(recv_buf);
+    int retry = 0;
+
+    while (retry < 10){
+        LogInfo("lwip select retry #: %d\n", retry);
+
+        FD_ZERO(&readset);
+        FD_SET(sock, &readset);
+    
+        FD_ZERO(&writeset);
+        FD_SET(sock, &writeset);
+    
+        FD_ZERO(&errset);
+        FD_SET(sock, &errset);
+    
+        ret = lwip_select(sock + 1, &readset, &writeset, &errset, NULL);
+        if (ret > 0){
+            if (FD_ISSET(sock, &writeset)){
+              break;
+            }
+    
+            if (FD_ISSET(sock, &readset)){
+                memset(recv_buf, 0, recv_bytes);
+                break;
+            }
+        }else{
+            LogInfo("lwip_select returned: %d\n", ret);
+        }
+
+        retry++;
+    }
+
+    int s = sock;
+    ctx = SSL_CTX_new(TLSv1_client_method());
+    if (!ctx) {
+        LogError("create new SSL CTX failed\n");
+        return;
+    }
+    LogInfo("create new SSL CTX OK\n");
+    
+    LogInfo("SSL set fragment\n");
+    ret = SSL_set_fragment(ctx, OPENSSL_FRAGMENT_SIZE);
```

Comment:
There is no propose to store the result of the SSL_set_fragment if you will not use it. Please consider to judge this result.